projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-admin-menu.php

```diff
@@ -370,8 +370,14 @@ private function prepare_menu_item_url( $url, $parent_slug = '' ) {
 				! file_exists( ABSPATH . "/wp-admin/$menu_file" )
 			)
 		) {
+			$admin_is_parent = false;
+			if ( ! empty( $parent_slug ) ) {
+				$menu_hook       = get_plugin_page_hook( $parent_slug, 'admin.php' );
+				$admin_is_parent = ! empty( $menu_hook ) || ( ( 'index.php' !== $parent_slug ) && file_exists( WP_PLUGIN_DIR . "/$parent_file" ) && ! file_exists( ABSPATH . "/wp-admin/$parent_file" ) );
+			}
+
 			if (
-				( 'admin.php' !== $parent_file && file_exists( WP_PLUGIN_DIR . "/$parent_file" ) && ! is_dir( WP_PLUGIN_DIR . "/$parent_file" ) ) ||
+				( false === $admin_is_parent && 'admin.php' !== $parent_file && file_exists( WP_PLUGIN_DIR . "/$parent_file" ) && ! is_dir( WP_PLUGIN_DIR . "/$parent_file" ) ) ||
```

Comment:
Miguel [had a suggestion](https://github.com/Automattic/jetpack/pull/21881/files#r763178687) to remove the `'admin.php' !== $parent_file`. Should we adopt that here?

Reply:
yes, forgot about that change, thanks!